azavea/raster-foundry
# Refactor color correction storage

## Overview

This refactors color correction storage to store each field as a separate object. So rather than one monolithic object like 
```javascript
{
    redMin: 0,
    redMax: 253,
    saturation: 1.3,
    ...
}
```

we will store this:
```javascript
{
    saturation: {
        enabled: true,
        saturation: 1.3
    },
    ...
}
```

This allows us to disable and reenable corrections without losing the values they store. It also simplified a lot of the frontend code because it more closely maps to how the interface actually works.

### Checklist


## Notes

- This is a good example of how we might migrate JSONB fields in the future.
- The color correction code could use further cleanup, both frontend and backend, but I was able to simplify a lot of it.

## Testing Instructions
This was a really wide-ranging PR that affected a lot of stuff, so it should be tested thoroughly; I'll be a bit surprised if this doesn't introduce any regressions.

 * Run `mg update` + `mg apply`
 * Navigate to an existing project with ingested scenes and color-correct the heck out of it. Confirm that everything still works and is persisted.
 * Enable / disable different color corrections and make sure they return to their previous values.
 * Change the color mode and confirm that it also works.

Closes #1644 
Closes #1603 
Apparently also closes #1837
